### DISCUSSION TITLE
Can't get HX-Push-Url to work

### DISCUSSION THREAD

**Discussion by lezhnev74** [bigskysoftware/htmx#1471] [Q&A] [answered]
Hey! I am new with the tool. And I can't get the browser history to update with a header `HX-Push-Url` (nor with `HX-Replace-Url`). I see the header (see the image) in the reponse from the server but the browser URL is not updated. Using an html attribute works (but lacks dynamic value as you know). So I am puzzled what am I doing wrong? Where do I look?

Note that I work in a local environment (if that matters).

Thank you for any tips :pray: 

![image](https://github.com/bigskysoftware/htmx/assets/10206110/f50829b6-6110-472b-ab7c-eb774a5a7584)

**Comment by andryyy** [comment, marked as answer]
Could it be a CORS issue?
Is the API in another location than the frontend?
You may need to allow the headers then.

**Reply by lezhnev74** [reply]
I see that a preflight request passes ok (which means no cors violation is detected by the server).  This header is sent from the server (not from a client), so should not violate any security, right? (as the server decides what to send back)
![image](https://github.com/bigskysoftware/htmx/assets/10206110/e79f5a90-ff0a-446d-837e-1b345f24a04e)

**Reply by jreviews** [reply]
Not sure it matters, but what if you try using `HX-Push-Url` instead of `Hx-Push-Url`

**Reply by andryyy** [reply]
And check Access-Control-Allow-Headers, it’s in the docs.

**Reply by lezhnev74** [reply]
After a few tests setting `access-control-expose-headers: *` made that working! 
So that was indeed related to CORS, thank you @andryyy for hints!
@jreviews good point too :pray:

**Comment by lezhnev74** [comment]
Here is a link to the docs that explain that: https://htmx.org/docs/#cors
